### ISSUE TITLE
Throwing IllegalStateException in event handler doesn't update event processor offset

### ISSUE THREAD

**Issue by MatteCarra** [lagom/online-auction-scala#40]
Throwing IllegalStateException in event handler doesn't update event offset and the event is handled over and over again.

**Comment by ignasi35** [comment]
hi @MatteCarra , I am assuming you refer to https://github.com/lagom/online-auction-scala/blob/24df3bb3d370a9d99f3b9250356abdc8fd91d621/item-impl/src/main/scala/com/example/auction/item/impl/ItemRepository.scala#L162.

Receiving an event for an `Item` which still doesn't have a creator should never happen and would be the symptom of another bug. You are right that throwing an exception would make the event processor to effectively stop processing(more on that later). The alternative would be to raise an alert for an admin to handle the situation and place the item auction on hold.
The code as is is prevents the read side from skipping one event if it can't be processed to avoid creating a corrupted read-side.

The actual outcome, though, is that only 1/4 of the events are blocked by this event processor not accepting on event since the event stream is sharded into [4 shards](https://github.com/lagom/online-auction-scala/blob/cea57d86231819f32126ee11360f54f2425e3088/item-impl/src/main/scala/com/example/auction/item/impl/ItemEntity.scala#L178-L178).

I think you raised an interesting point and I'm looking forward to discussing alternative ways to improve this part of the `online-auction` sample app.

**Comment by TimMoore** [comment]
I'm not sure if there's anything we want to change here. @MatteCarra did you encounter a situation where the application actually got into this illegal state?

**Comment by TimMoore** [comment]
Closing due to no response from the original submitter.
